internal/k8s/controllers/pool_controller.go

```diff
@@ -83,7 +84,19 @@ func (r *PoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 
 	level.Debug(r.Logger).Log("controller", "PoolReconciler", "metallb CRs", dumpClusterResources(&resources))
 
-	cfg, err := config.For(resources, r.ValidateConfig)
+	cfg, err := toConfig(r.ValidateConfig,
+		ipAddressPools.Items,
+		[]metallbv1beta2.BGPPeer{},
+		[]metallbv1beta1.BFDProfile{},
+		[]metallbv1beta1.L2Advertisement{},
+		[]metallbv1beta1.BGPAdvertisement{},
+		addressPools.Items,
+		communities.Items,
+		map[string]corev1.Secret{},
+		[]corev1.Node{},
+		namespaces.Items,
+		corev1.ConfigMap{})
```

Comment:
That's a little hard to read, plus we'll need to update the parameters list when a new CRD is added. why not reuse the `ClusterResources` struct, consume it as the input, and re-assign its fields with the sorted copy?
I would go even a step further and split `toConfig` into two methods, first only the logic of sorting the resources, so we'll have a `sortedResources`, and then call the `config.For` on it.

Reply:
> That's a little hard to read, plus we'll need to update the parameters list when a new CRD is added. 

What I was totally afraid of was to forget the sorting in case a new CRD is added. I wanted to make it really hard to forget, and that's why I came up with the long list. But I guess I can fix that by doing the sorting in the toConfig as you said. Let me try to get a stab at it